lib/libc/gen/getgrent.c

```diff
@@ -373,9 +373,9 @@ grp_unmarshal_func(char *buffer, size_t buffer_size, void *retval, va_list ap,
 
 	orig_buf = (char *)_ALIGN(orig_buf);
 	memcpy(orig_buf, buffer + sizeof(struct group) + sizeof(char *) +
-	    (vaddr_t)_ALIGN(p) - (vaddr_t)p,
+	    (ptraddr_t)_ALIGN(p) - (ptraddr_t)p,
```

Comment:
I suspect strictly speaking this should be `(char *)_ALIGN(p) - p` to avoid temporary OOB issues, and that comes with the advantage that you get a ptrdiff_t without any need to cast... (and this pattern is repeated a bunch for similar files)

Reply:
In CheriBSD (at least on CHERI architectures) the `(char *)` casts are gratuitous since `__builtin_align_up` is type preserving, but upstream call has `_ALIGN` returning`long`. I'll go ahead and add them in a cleanup.

We should probably upstream the `_ALIGN` change, but it likely requires a ports exp-run since it's an API change.